### ISSUE TITLE
Refocusing the text input field with a mouse click without confirming text breaks input until I unfocus/refocus with keyboard in Firefox

### ISSUE THREAD

**Issue by konomu** [fcitx/fcitx5#1080]
### Summary

This issue happens to me with Firefox 127.0.

 If I start writing Japanese and then unfocus/refocus the text field with left mouse click without pressing Enter to confirm the text, the next thing I type won't show up in the text field. When this happens, everything I type will show up in the candidate window, but will not actually be written to the text field even if I press enter on a candidate. 

 The issue will not go away unless I press Esc to delete what I wrote and exit out of the candidate window, or backspace the characters that I wrote without pressing Enter.

### Steps to Reproduce

1. Switch to Mozc input method
2. Start writing in any text field in Firefox (127.0)
3. Don't press Enter to confirm text, click on the text field
4. Start writing again.

### Expected Behavior

The text should appear in the text field as I'm writing, but nothing appears. Instead, the text only appears in the candidate window and can't be written to the text field. I can only write again when I press Esc, switch input method, or delete the characters that were unconfirmed when I left-clicked.

### Output of fcitx5-diagnose command

# System Info:
1.  `uname -a`:

        Linux pasocon 6.8.0-35-generic #35-Ubuntu SMP PREEMPT_DYNAMIC Mon May 20 15:51:52 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

2.  `lsb_release -a`:

        Distributor ID:	Ubuntu
        Description:	Ubuntu 24.04 LTS
        Release:	24.04
        Codename:	noble

3.  `lsb_release -d`:

        Description:	Ubuntu 24.04 LTS

4.  `/etc/lsb-release`:

        DISTRIB_ID=Ubuntu
        DISTRIB_RELEASE=24.04
        DISTRIB_CODENAME=noble
        DISTRIB_DESCRIPTION="Ubuntu 24.04 LTS"

5.  `/etc/os-release`:

        PRETTY_NAME="Ubuntu 24.04 LTS"
        NAME="Ubuntu"
        VERSION_ID="24.04"
        VERSION="24.04 LTS (Noble Numbat)"
        VERSION_CODENAME=noble
        ID=ubuntu
        ID_LIKE=debian
        HOME_URL="https://www.ubuntu.com/"
        SUPPORT_URL="https://help.ubuntu.com/"
        BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
        PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
        UBUNTU_CODENAME=noble
        LOGO=ubuntu-logo

6.  Desktop Environment:

    Desktop environment is `gnome3`.

7.  XDG SESSION TYPE:

        XDG_SESSION_TYPE='x11'

8.  Bash Version:

        BASH_VERSION='5.2.21(1)-release'

# Environment:
1.  DISPLAY:

        DISPLAY=':1'

        WAYLAND_DISPLAY=''

2.  Keyboard Layout:

    1.  `setxkbmap`:

            xkb_keymap {
            	xkb_keycodes  { include "evdev+aliases(qwerty)"	};
            	xkb_types     { include "complete"	};
            	xkb_compat    { include "complete"	};
            	xkb_symbols   { include "pc+us+inet(evdev)"	};
            	xkb_geometry  { include "pc(pc104)"	};
            };

    2.  `xprop`:

            _XKB_RULES_NAMES(STRING) = "evdev", "pc105+inet", "us", "", ""

3.  Locale:

    1.  All locales:

            C
            C.utf8
            POSIX
            en_AG
            en_AG.utf8
            en_AU.utf8
            en_BW.utf8
            en_CA.utf8
            en_DK.utf8
            en_GB.utf8
            en_HK.utf8
            en_IE.utf8
            en_IL
            en_IL.utf8
            en_IN
            en_IN.utf8
            en_NG
            en_NG.utf8
            en_NZ.utf8
            en_PH.utf8
            en_SG.utf8
            en_US.utf8
            en_ZA.utf8
            en_ZM
            en_ZM.utf8
            en_ZW.utf8
            ja_JP.utf8

    2.  Current locale:

            LANG=ja_JP.UTF-8
            LANGUAGE=ja:en_US:en
            LC_CTYPE="C"
            LC_NUMERIC="C"
            LC_TIME="C"
            LC_COLLATE="C"
            LC_MONETARY="C"
            LC_MESSAGES="C"
            LC_PAPER="C"
            LC_NAME="C"
            LC_ADDRESS="C"
            LC_TELEPHONE="C"
            LC_MEASUREMENT="C"
            LC_IDENTIFICATION="C"
            LC_ALL=C

4.  Directories:

    1.  Home:

            /home/konomu

    2.  `${XDG_CONFIG_HOME}`:

        Environment variable `XDG_CONFIG_HOME` is not set.

        Current value of `XDG_CONFIG_HOME` is `~/.config` (`/home/konomu/.config`).

    3.  Fcitx5 Settings Directory:

        Current fcitx5 settings directory is `~/.config/fcitx5` (`/home/konomu/.config/fcitx5`).

5.  Current user:

    The script is run as konomu (1000).

# Fcitx State:
1.  executable:

    Found fcitx5 at `/usr/bin/fcitx5`.

2.  version:

    Fcitx version: `5.1.7`

3.  process:

    Found 3 fcitx5 processes:

        1897703 fcitx5
        2387943 fcitx5-config-q
        3699225 fcitx5-config-q

4.  `fcitx5-remote`:

    `fcitx5-remote` works properly.

5.  DBus interface:

    Using `dbus-send` to check dbus.

    Owner of DBus name `org.fcitx.Fcitx5` is `:1.66`.

    PID of DBus name `org.fcitx.Fcitx5` owner is `1897703`.

    Debug information from dbus:

           Group [x11::1] has 32 InputContext(s)
          IC [c0a864dc593341a29d3a4cd7c657a155] program:fcitx5-config-qt frontend:dbus cap:16000800072 focus:0
          IC [73e43478da9b47a4b0372c8b06821cb0] program:Discord frontend:dbus cap:6000000032 focus:0
          IC [77e33a86993241378cf74e6f40fba5ff] program:xdg-desktop-portal-gnome frontend:dbus cap:4000000032 focus:0
          IC [3d28e54c3d014651b9e97ed44ca36c54] program:firefox frontend:fcitx4 cap:12 focus:0
          IC [34a73a8a26c64721830e6d469ea7738a] program:firefox frontend:fcitx4 cap:12 focus:0
          IC [5684b6ffa8904b0f9d39c33f0e7d5cd1] program:firefox frontend:fcitx4 cap:12 focus:0
          IC [c77941d12a494d499684e6ad54830c11] program:xdg-desktop-portal-gnome frontend:dbus cap:4000000032 focus:0
          IC [fac79fcee6cd4ee295b1a0ffcefa6dae] program:firefox frontend:fcitx4 cap:12 focus:0
          IC [51182e3700694c5bb547992241753d93] program:firefox frontend:fcitx4 cap:12 focus:0
          IC [ac0bbc5c1d7b4efbac1d8000f1878619] program:firefox frontend:fcitx4 cap:12 focus:0
          IC [15b5932e100848698d37e8805dde3ba7] program:firefox frontend:fcitx4 cap:12 focus:0
          IC [6d052ad497dc4685ad09ccb738eaaa3a] program:firefox frontend:fcitx4 cap:12 focus:0
          IC [ea7f2be280424b9da5b67b048f44a90b] program:fcitx5-config-qt frontend:dbus cap:16000800072 focus:0
          IC [be8176317a764d14a85ccdb098dfa55e] program:firefox frontend:fcitx4 cap:52 focus:0
          IC [400e2db96b494e66a24adbe68c7a8565] program:firefox frontend:fcitx4 cap:12 focus:0
          IC [9087a77a6c2b4e3985ad6a590e467fc0] program: frontend:ibus cap:52 focus:0
          IC [e340961267ad474c95da1853322581d8] program:gnome-software frontend:dbus cap:4000000032 focus:0
          IC [13c39b991f434d5d8f8e9970bb5211af] program:gnome-terminal-server frontend:dbus cap:6000000032 focus:1
          IC [2922fb2682794191988e57d350a4fcc0] program:firefox frontend:fcitx4 cap:12 focus:0
          IC [5740a29023674177be7999323a022779] program:firefox frontend:fcitx4 cap:12 focus:0
          IC [ac03110a728f49aa9b67bd772b3d74dd] program:xdg-desktop-portal-gnome frontend:dbus cap:4000000032 focus:0
          IC [5c0ce6de080344228e9e824bd02f624c] program:xdg-desktop-portal-gnome frontend:dbus cap:4000000032 focus:0
          IC [e4bc7ef00f0544eaa3430f8c98ec889d] program:xdg-desktop-portal-gnome frontend:dbus cap:4000000032 focus:0
          IC [96fb601309a54b7e915aa91269fcfdbe] program:firefox frontend:fcitx4 cap:12 focus:0
          IC [fc6fdbfae38943af8ae0e5183e74476c] program:xdg-desktop-portal-gnome frontend:dbus cap:4000000032 focus:0
          IC [de2806e54756465ebe144c6c69ee00e5] program:firefox frontend:fcitx4 cap:12 focus:0
          IC [6378bc6c53544440840f153f3e94538f] program:firefox frontend:fcitx4 cap:12 focus:0
          IC [953adfa13ce147ab93bd7dee83a1a7c5] program:xdg-desktop-portal-gnome frontend:dbus cap:4000000032 focus:0
          IC [4f758142d0aa44b3a32afb8f497e4ca6] program:xdg-desktop-portal-gnome frontend:dbus cap:4000000032 focus:0
          IC [fd2646ce8da5448c9e5407693584f9db] program:firefox frontend:fcitx4 cap:12 focus:0
          IC [a64b743d5e354239a51fd18f54ff00e1] program:firefox frontend:fcitx4 cap:12 focus:0
          IC [6ebd3170741e4c24a321d889507c14d1] program:firefox frontend:fcitx4 cap:12 focus:0
        Input Context without group

# Fcitx Configure UI:
1.  Config Tool Wrapper:

    Found fcitx5-configtool at `/usr/bin/fcitx5-configtool`.

2.  Config GUI for qt:

    Found `fcitx5-config-qt` at `/usr/bin/fcitx5-config-qt`.

3.  Config GUI for kde:

    **`kcmshell5` not found.**

# Frontends setup:
The environment variable checked by this script only shows the environment under current shell. It is still possible that you did not set the environment to the whole graphic desktop session. You may inspect the actual environment variable of a certain process by using `xargs -0 -L1 /proc/$PID/environ` for a certain process that you find not working.

## Xim:
1.  `${XMODIFIERS}`:

    Environment variable XMODIFIERS is set to "@im=fcitx" correctly.
    Xim Server Name from Environment variable is fcitx.

2.  XIM_SERVERS on root window:

    **Xim server name: "ibus" is different from that set in the environment variable: "fcitx".**

3.  XIM encoding:

    **Your LC_CTYPE is set to C whose encoding is not UTF-8. You may have trouble committing strings using XIM.**

## Qt:
1.  qt4 - `${QT4_IM_MODULE}`:

    Environment variable QT_IM_MODULE is set to "fcitx" correctly.

    **`fcitx5-qt4-immodule-probing` not found.**

2.  qt5 - `${QT_IM_MODULE}`:

    Environment variable QT_IM_MODULE is set to "fcitx" correctly.

    Using fcitx5-qt5-immodule-probing to check the actual im module to be used under current environment:

        QT_QPA_PLATFORM=xcb
        QT_IM_MODULE=fcitx
        IM_MODULE_CLASSNAME=fcitx::QFcitxPlatformInputContext

3.  qt6 - `${QT_IM_MODULE}`:

    Environment variable QT_IM_MODULE is set to "fcitx" correctly.

    Using fcitx5-qt6-immodule-probing to check the actual im module to be used under current environment:

        QT_QPA_PLATFORM=xcb
        QT_IM_MODULE=fcitx
        IM_MODULE_CLASSNAME=fcitx::QFcitxPlatformInputContext

4.  Qt IM module files:

    Found fcitx5 im module for qt6: `/lib/x86_64-linux-gnu/qt6/plugins/platforminputcontexts/libfcitx5platforminputcontextplugin.so`.
    Found fcitx5 qt6 module: `/lib/x86_64-linux-gnu/fcitx5/qt6/libfcitx-quickphrase-editor5.so`.
    Found fcitx5 im module for qt5: `/lib/x86_64-linux-gnu/qt5/plugins/platforminputcontexts/libfcitx5platforminputcontextplugin.so`.

    Following error may not be accurate because guessing Qt version from path depends on how your distribution packages Qt. It is not a critical error if you do not use any Qt application with certain version of Qt or you are using text-input support by Qt under Wayland.
    **Cannot find fcitx5 input method module for Qt4.**

## Gtk:
1.  gtk - `${GTK_IM_MODULE}`:

    Environment variable GTK_IM_MODULE is set to "fcitx" correctly.

    **`fcitx5-gtk2-immodule-probing` not found.**

    Using fcitx5-gtk3-immodule-probing to check the actual im module to be used under current environment:

        GTK_IM_MODULE=fcitx

    Using fcitx5-gtk4-immodule-probing to check the actual im module to be used under current environment:

        GTK_IM_MODULE=fcitx

2.  `gtk-query-immodules`:

    1.  gtk 2:

        **Cannot find `gtk-query-immodules` for gtk 2**

        **Cannot find fcitx5 im module for gtk 2.**

    2.  gtk 3:

        **Cannot find `gtk-query-immodules` for gtk 3**

        **Cannot find fcitx5 im module for gtk 3.**

3.  Gtk IM module cache:

    1.  gtk 2:

        Found immodules cache for gtk `2.24.33` at `/lib/x86_64-linux-gnu/gtk-2.0/2.10.0/immodules.cache`.
        Version Line:

            # Created by /usr/lib/x86_64-linux-gnu/libgtk2.0-0t64/gtk-query-immodules-2.0 from gtk+-2.24.33

        **Failed to find fcitx5 in immodule cache at `/lib/x86_64-linux-gnu/gtk-2.0/2.10.0/immodules.cache`**

        **Cannot find fcitx5 im module for gtk 2 in cache.**

    2.  gtk 3:

        Found immodules cache for gtk `3.24.41` at `/lib/x86_64-linux-gnu/gtk-3.0/3.0.0/immodules.cache`.
        Version Line:

            # Created by /usr/lib/x86_64-linux-gnu/libgtk-3-0t64/gtk-query-immodules-3.0 from gtk+-3.24.41

        Found fcitx5 im modules for gtk `3.24.41`.

            "/usr/lib/x86_64-linux-gnu/gtk-3.0/3.0.0/immodules/im-fcitx5.so" 
            "fcitx" "Fcitx5 (Flexible Input Method Framework5)" "fcitx5" "/usr/locale" "ja:ko:zh:*" 
            "fcitx5" "Fcitx5 (Flexible Input Method Framework5)" "fcitx5" "/usr/locale" "ja:ko:zh:*" 

4.  Gtk IM module files:

    1.  gtk 2:

        All found Gtk 2 immodule files exist.

    2.  gtk 3:

        All found Gtk 3 immodule files exist.

    3.  gtk 4:

        All found Gtk 4 immodule files exist.

# Configuration:
## Fcitx Addons:
1.  Addon Config Dir:

    Found fcitx5 addon config directory: `/usr/share/fcitx5/addon`.

2.  Addon List:

    1.  Found 21 enabled addons:

            Classic User Interface 5.1.7
            Clipboard 5.1.7
            DBus 5.1.7
            DBus Frontend 5.1.7
            Emoji 5.1.7
            Fcitx4 Frontend 5.1.7
            IBus Frontend 5.1.7
            Input method selector 5.1.7
            Keyboard 5.1.7
            KDE Input Method Panel 5.1.7
            Mozc 
            Status Notifier 5.1.7
            Notification 5.1.7
            Quick Phrase 5.1.7
            Spell 5.1.7
            Unicode 5.1.7
            DBus Virtual Keyboard 5.1.7
            Wayland 5.1.7
            Wayland Input method frontend 5.1.7
            XCB 5.1.7
            X Input Method Frontend 5.1.7

    2.  Found 0 disabled addons:

3.  Addon Libraries:

    All libraries for all addons are found.

4.  User Interface:

    Found 3 enabled user interface addons:

        Classic User Interface
        KDE Input Method Panel
        DBus Virtual Keyboard

## Input Methods:
1.  `/home/konomu/.config/fcitx5/profile`:

        [Groups/0]
        # Group Name
        Name=Default
        # Layout
        Default Layout=us
        # Default Input Method
        DefaultIM=mozc

        [Groups/0/Items/0]
        # Name
        Name=keyboard-us
        # Layout
        Layout=

        [Groups/0/Items/1]
        # Name
        Name=mozc
        # Layout
        Layout=

        [Groups/0/Items/2]
        # Name
        Name=keyboard-jp-kana
        # Layout
        Layout=

        [GroupOrder]
        0=Default

# Log:
1.  `date`:

        Sun Jun 23 18:39:04 PDT 2024

2.  `/home/konomu/.config/fcitx5/crash.log`:

    `/home/konomu/.config/fcitx5/crash.log` not found.

**Comment by konomu** [comment]
This appears to be a Mozc-related issue, so I'll close this issue.

**Comment by wengxt** [comment]
@konomu  No, this is not a mozc issue.

So basically , I can see you're using snap version of firefox, which can only work with fcitx4 im module.

But in fcitx5's gtk im module we already monitor the mouse click event and commit the text:
https://github.com/fcitx/fcitx5-gtk/blob/cb9aee2f2b5f710ffaaa2a65d4bbf0df32e50947/gtk3/fcitximcontext.cpp#L469

I'm not closely follow how ubuntu package snap version and I know there at least used to be some other issue that only package fcitx4 im module and block the fcitx5 dbus interface via apparmor.

In short, I believe if you use flatpak firefox, or deb firefox, the issue would be gone.
